File: solutions/nim/darts/1/darts.nim
import std/math

proc score*(x, y: float): int =
  let distance = sqrt(x*x + y*y)
  case distance:
    of 0.0 .. 1.0:
      10
    of 1.001 .. 5.0:
      5
    of 5.001 .. 10.0:
      1
    else:
      0

File: solutions/ocaml/darts/1/darts.ml
let score (x: float) (y: float): int =
  let distance = sqrt (x *. x +. y *. y) in
  match distance with
  | _ when distance <= 1.0 -> 10
  | _ when distance <= 5.0 -> 5
  | _ when distance <= 10.0 -> 1
  |_ -> 0
